src/phoebus_driver.cpp

```diff
@@ -1014,11 +1017,211 @@ TaskListStatus PhoebusDriver::MonteCarloStep() {
+        const Real vp1[3] = {v(0, p::velocity(0), k, j, i - 1),
+                             v(0, p::velocity(1), k, j, i - 1),
+                             v(0, p::velocity(2), k, j, i - 1)};
+        geom.Metric(CellLocation::Cent, 0, k, j, i, gam);
+        geom.Metric(CellLocation::Cent, 0, k, j, i - 1, gam1);
+        Real gdet = geom.DetGamma(CellLocation::Cent, 0, k, j, i);
+        const Real W = phoebus::GetLorentzFactor(vp, gam);
+        const Real W1 = phoebus::GetLorentzFactor(vp1, gam1);
+
+        if (ndim == 1) {
+          divv = 1 / gdet *
+                 (geom.DetGamma(CellLocation::Cent, 0, k, j, i) *
+                      v(0, p::velocity(0), k, j, i) / W -
+                  geom.DetGamma(CellLocation::Cent, 0, k, j, i - 1) *
+                      v(0, p::velocity(0), k, j, i - 1) / W1) /
+                 coords.CellWidthFA(X1DIR, k, j, i);
+        }
+
+        if (ndim == 2) {
+          Real vp2[3] = {v(0, p::velocity(0), k, j - 1, i),
+                         v(0, p::velocity(1), k, j - 1, i),
+                         v(0, p::velocity(2), k, j - 1, i)};
+          Real vp12[3] = {v(0, p::velocity(0), k, j - 1, i - 1),
+                          v(0, p::velocity(1), k, j - 1, i - 1),
+                          v(0, p::velocity(2), k, j - 1, i - 1)};
+          geom.Metric(CellLocation::Cent, 0, k, j - 1, i, gam2);
+          geom.Metric(CellLocation::Cent, 0, k, j - 1, i - 1, gam12);
+          Real W2 = phoebus::GetLorentzFactor(vp2, gam2);
+          Real W12 = phoebus::GetLorentzFactor(vp12, gam12);
+
+          divv = 1 / gdet *
+                     (geom.DetGamma(CellLocation::Cent, 0, k, j, i) *
+                          v(0, p::velocity(0), k, j, i) / W -
+                      geom.DetGamma(CellLocation::Cent, 0, k, j, i - 1) *
+                          v(0, p::velocity(0), k, j, i - 1) / W1 +
+                      geom.DetGamma(CellLocation::Cent, 0, k, j - 1, i) *
+                          v(0, p::velocity(0), k, j - 1, i) / W2 -
+                      geom.DetGamma(CellLocation::Cent, 0, k, j - 1, i - 1) *
+                          v(0, p::velocity(0), k, j - 1, i - 1) / W12) /
+                     coords.CellWidthFA(X1DIR, k, j, i) / 2
+
+                 + 1 / gdet *
+                       (geom.DetGamma(CellLocation::Cent, 0, k, j, i) *
+                            v(0, p::velocity(1), k, j, i) / W -
+                        geom.DetGamma(CellLocation::Cent, 0, k, j - 1, i) *
+                            v(0, p::velocity(1), k, j - 1, i) / W2 +
+                        geom.DetGamma(CellLocation::Cent, 0, k, j, i - 1) *
+                            v(0, p::velocity(1), k, j, i - 1) / W1 -
+                        geom.DetGamma(CellLocation::Cent, 0, k, j - 1, i - 1) *
+                            v(0, p::velocity(1), k, j - 1, i - 1) / W12) /
+                       coords.CellWidthFA(X2DIR, k, j, i) / 2;
+        }
+
+        if (ndim == 3) {
```

Comment:
These expressions can be much simplified by accumulating `divv` as you walk up in dimension. Fr example (simplified):
```C++
// compute once to minimize number of divisions
// also use phoebus_utils/robust.hpp to guard against division by 0
Real igdet = robust::ratio(1., gdet); 
divv = igdet * (gdet * v(0, p::velocity(1), k, j, i) / W - geom.DEtGamma(CellLocation::Cent, 0, k, j, i-1)*v(0, p::velocity(1), k, j, i-1) /W1) * robust::ratio(alpha, coords.CellWidthFA(X1DIR, k, j, i-1))
if (ndim >= 2) {
  // compute relevant things...
  divv += igdet * (gdet * v(0, p::velocity(2), k, j, i) / W - geom.DEtGamma(CellLocation::Cent, 0, k, j-1, i)*v(0, p::velocity(2), k, j-1, i) /W2) * robust::ratio(alpha, coords.CellWidthFA(X2DIR, k, j-1, i))
}
if (ndim >= 3) {
  // similar thing
}